### PR TITLE
[BTS-1172] Backports to 3.10

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

Mostly fixes for sanitizers

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1195
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Comment by MBkkt** [comment]
https://github.com/arangodb/enterprise/pull/1143

Also it backport this, not intentional, but I think it's good.

Don't backport main PR and script (will be backported if jenkins will failed)

**Review by gnusi — APPROVED** [review]

**Review by Dronplane — APPROVED** [review]
LGTM

File: lsan_arangodb_suppressions.txt [hunk 1]
```diff
@@ -17,3 +17,7 @@ leak:snowball
 #  4 in v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) 
 # SUMMARY: AddressSanitizer: 139 byte(s) leaked in 2 allocation(s).
 leak:V8Buffer::New
+
+# Suppressed to see other issue
+# TODO(mpoeter) ?
+leak:LogContext::Current::appendEntry
```
Comment (jsteemann) [inline lsan_arangodb_suppressions.txt:23]: Not necessary in 3.9. replication2 code doesn't exist in this branch.

Comment (MBkkt) [reply lsan_arangodb_suppressions.txt:23]: But it's from RestServer...?

Comment (MBkkt) [reply lsan_arangodb_suppressions.txt:23]: Anyway, it doesn't make it worse

File: 3rdParty/s2geometry/0e7b146/src/s2/util/gtl/compact_array.h [hunk 2]
```diff
@@ -413,8 +413,10 @@ class compact_array_base {
     value_allocator_type allocator;
 
     T* new_ptr = allocator.allocate(capacity());
-    memcpy(new_ptr, Array(), old_capacity * sizeof(T));
-    allocator.deallocate(Array(), old_capacity);
+    if (old_capacity == 0) {
```
Comment (jsteemann) [inline 3rdParty/s2geometry/0e7b146/src/s2/util/gtl/compact_array.h:416]: I don't understand this change. Shouldn't it be `if (old_capacity != 0)` here?

File: lsan_arangodb_suppressions.txt (hunk 1 above)
Comment (jsteemann) [reply lsan_arangodb_suppressions.txt:23]: LogContext is from replication2, isn't it?

Comment (MBkkt) [reply lsan_arangodb_suppressions.txt:23]: https://github.com/arangodb/arangodb/blob/3.10/lib/Logger/LogContext.h

No, it's other log, some Manuel experiment with structured logging if I remember correct

File: 3rdParty/s2geometry/0e7b146/src/s2/util/gtl/compact_array.h (hunk 2 above)
Comment (MBkkt) [reply 3rdParty/s2geometry/0e7b146/src/s2/util/gtl/compact_array.h:416]: Thanks!

File: lsan_arangodb_suppressions.txt (hunk 1 above)
Comment (jsteemann) [reply lsan_arangodb_suppressions.txt:23]: Ah, you are right.

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by KVS85 — APPROVED** [review]

**Pull request merged**